apis/projectcontour/v1/helpers.go

```diff
@@ -0,0 +1,79 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+// AuthorizationConfigured returns whether authorization  is
+// configured on this virtual host.
+func (v *VirtualHost) AuthorizationConfigured() bool {
+	return v.TLS != nil && v.Authorization != nil
+}
+
+// DisableAuthorization returns true if this virtual host disables
+// authorization. If an authorization server is present, the default
+// policy is to not disable.
+func (v *VirtualHost) DisableAuthorization() bool {
+	// No authorization, so it is disabled.
+	if v.AuthorizationConfigured() {
+		// No policy specified, default is to not disable.
+		if v.Authorization.AuthPolicy == nil {
+			return false
+		}
+
+		return v.Authorization.AuthPolicy.Disabled
+	}
+
+	return false
+}
+
+// AuthorizationContext returns the authorization policy context (if present).
+func (v *VirtualHost) AuthorizationContext() map[string]string {
+	if v.AuthorizationConfigured() {
+		if v.Authorization.AuthPolicy != nil {
+			return v.Authorization.AuthPolicy.Context
+		}
+	}
+
+	return nil
+}
+
+// GetPrefixReplacements returns replacement prefixes from the path
+// rewrite policy (if any).
+func (r *Route) GetPrefixReplacements() []ReplacePrefix {
+	if r.PathRewritePolicy != nil {
+		return r.PathRewritePolicy.ReplacePrefix
+	}
+	return nil
+}
+
+// AuthorizationContext merges the parent context entries with the
+// context from this Route. The parent map may be nil.
```

Comment:
Nit: Maybe something here about how the more specific context will override? Just to make it really clear?

Reply:
Fixed.